clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -832,6 +834,7 @@ public void onSuccess(OffsetsForLeaderEpochClient.OffsetForEpochResult offsetsRe
                     });
 
                     if (!truncationWithoutResetPolicy.isEmpty()) {
+                        log.error("Detected log truncation with diverging offsets " + truncationWithoutResetPolicy);
```

Comment:
As in the other case, can we make this info? It would also be helpful if we could include the request offset and epoch. Potentially we could move this log line into the loop above where we already access `requestPosition`.

Also nit: use `{}`